sst/sstable.go

```diff
@@ -161,18 +169,30 @@ func BuildSSTable(format common.DataFormat, buffSizeEstimate int, entriesEstimat
 		indexOffset:      uint32(indexOffset),
 		creationTime:     uint64(time.Now().UTC().UnixMilli()),
 		data:             buff,
-	}, smallestKey, largestKey, minVersion, maxVersion, nil
+	}
+
+	selfTable.metadataBinary = selfTable.serializedMetdata()
+	selfTable.metadataSize = len(selfTable.metadataBinary)
+
+	return selfTable, smallestKey, largestKey, minVersion, maxVersion, nil
+}
+
+func (s *SSTable) serializedMetdata() []byte {
```

Comment:
This method is not necessary and will cause extra unnecessary buffer copying. Simpler to just append the varints directly to the existing buffer.